src/net.cpp

```diff
@@ -1548,6 +1548,27 @@ void CConnman::QueryFixedSeeds(std::chrono::microseconds start)
                 seed_addrs.erase(std::remove_if(seed_addrs.begin(), seed_addrs.end(),
                                                 [&fixed_seed_networks](const CAddress& addr) { return fixed_seed_networks.count(addr.GetNetwork()) == 0; }),
                                  seed_addrs.end());
+                Shuffle(seed_addrs.begin(), seed_addrs.end(), FastRandomContext());
+
+                // Make AddrFetch connections to fixed seeds first. This reduces the
+                // load on the fixed seeds that would otherwise be serving blocks for
+                // many new peers requiring IBD. Try this with multiple fixed seeds for
+                // a better diversity of received addrs and because some may be offline.
+                LogPrintf("Initiating AddrFetch connections to fixed seeds.\n");
+                for (size_t addr_pos = 0; addr_pos < 10; ++addr_pos) {
+                    if (addr_pos >= seed_addrs.size()) {
+                        break;
+                    }
+                    AddAddrFetch(seed_addrs.at(addr_pos).ToStringAddr());
+                }
+                // Give AddrFetch peers some time to provide us with addresses
+                // before adding the fixed seeds to AddrMan
+                if (!interruptNet.sleep_for(std::chrono::minutes(2))) {
```

Comment:
30b0f57b19eddd47fa137d0c20ae39174b54dbad

Doesn't this substantially change the observations of a user? If they get unlucky with the 10 seeds and gets stuck for 2 minutes, which was not the case before?

Maybe worth expanding a log `this might take up to 2 minutes.` or something. Ideally we won't wait anymore once we realize all those attempts failed already, but that might be difficult to code.

Reply:
I think that usually it shouldn't substantially change the observations of a user, because 10 is a pretty high number and if all of those fail, we either have rally bad fixed seeds, or have a connectivity problem. I'll add a log entry with the next push though.
Also, I'll do some test runs to gather some statistics whether we might want to reduce this to 1 minute...

Reply:
added the log.

Reply:
nit: the log might use a variable `2`, so that it's defined in a single place.

Reply:
> I think that usually it shouldn't substantially change the observations of a user, because 10 is a pretty high number and if all of those fail, we either have rally bad fixed seeds, or have a connectivity problem.

I agree. I've tested it following `contrib/seeds/README.md` to generate the seeds and the worst scenario was connecting with 5 of the 10.

Reply:
> I agree. I've tested it following `contrib/seeds/README.md` to generate the seeds and the worst scenario was connecting with 5 of the 10.

Yes, I see similar results. If older versions are used, it could become a bit worse though.

> nit: the log might use a variable 2, so that it's defined in a single place.

I have now used a variable for the 2 minutes (sorry, I missed this comment before).